### ISSUE TITLE
Function fit skeleton

### ISSUE THREAD

**Issue by saurav193** [UBC-MDS/Kmeans_python#5]
Docstring and skeleton only

**Comment by RobBlumberg** [comment]
Because we are not writing all the functions in a single class, the fit function will need to output the cluster centers and the assigned cluster for each point in the training set.

**Comment by sreejithmunthikodu** [comment]
We also need inertia from fit function.

**Comment by RobBlumberg** [comment]
Total inertia or within-cluster inertia? The summary function will show within-cluster inertia, so let's make sure we're not calculating this twice. I was thinkning to do the intertia calculation in the summary function to make that function more involved, since fit will already be decently involved. so fit would be just fitting, and summary would be kind of like a fancier "score" method

**Comment by sreejithmunthikodu** [comment]
I was referring to sum of intra-cluster distances. I guess this is what we will be minimizing in the fit function. So we don't have to calculate this again in the summarize function. elbow function will be calling the fit function for each $K$ to get the inertia. Otherwise, it will be redundant computation.

**Comment by RobBlumberg** [comment]
I think when determining the clusters in k-means, we don't directly minimize a loss, we just reassign points to groups based on distance and recalculate centroids iteratively until we no longer need to reassign points

**Comment by sreejithmunthikodu** [comment]
Should we add an option for the type of distance metric? I guess it will only be an additional 2 lines of code. 
1. scipy.spatial.distance.mahalanobis
2. scipy.spatial.distance.euclidean

**Comment by RobBlumberg** [comment]
Sure,  good idea, we can also add manhattan distance. This can be a kwarg in the fit function

**Comment by sreejithmunthikodu** [comment]
> I think when determining the clusters in k-means, we don't directly minimize a loss, we just reassign points to groups based on distance and recalculate centroids iteratively until we no longer need to reassign points

True.. But we are calculating the distance anyways inside the fit function. For inertia, it only needs to sum at the end based on final cluster, which is not expensive. I prefer it inside the fit function because otherwise, elbow function will have to call the summarize function to get the inertia. The intuition is, summarize function kind of summarizes everything at the end.What do you guys think?

**Comment by RobBlumberg** [comment]
You're right i agree, i guess this means that fit should output within cluster distances and the summarize function will have the within cluster distances as input so that we don't have to recalculate it

**Comment by sreejithmunthikodu** [comment]
I thought summarize function would call fit to get the centroids. So it will get inertia too without having to explicitly feed in. Is it not the case?

**Comment by RobBlumberg** [comment]
I guess I was thinking summarize would receive all the outputs from fit instead of internally calling fit. My thought behind this was that if I call fit on the data, and then later I decide I want to see the summary, I wouldn't want to call fit again? I also thought this would make it easier in case we decide to put everything in a class later

**Comment by sreejithmunthikodu** [comment]
I agree. I was thinking of fewer input arguments to the function for ease of use.

**Comment by RobBlumberg** [comment]
To summarize:

input to fit() should be:
- data
- k 
- everything else should be kwarg (ex: distance_metric)

output should be
- cluster center locations (numpy.ndarray)
- cluster assignements for training data  (array-like)

agreed?

**Comment by saurav193** [comment]
For now I am not adding the **kwarg argument. If the functions tests ok, I will add it next week.  Lets keep 'euclidean' for now.
